pkg/apis/flows/v1beta1/parallel_types.go

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright 2019 The Knative Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	eventingduckv1alpha1 "knative.dev/eventing/pkg/apis/duck/v1alpha1"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	duckv1alpha1 "knative.dev/pkg/apis/duck/v1alpha1"
+	"knative.dev/pkg/apis/v1alpha1"
+	"knative.dev/pkg/kmeta"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// Parallel defines conditional branches that will be wired in
+// series through Channels and Subscriptions.
+type Parallel struct {
+	metav1.TypeMeta `json:",inline"`
+	// +optional
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines the desired state of the Parallel.
+	Spec ParallelSpec `json:"spec,omitempty"`
+
+	// Status represents the current state of the Parallel. This data may be out of
+	// date.
+	// +optional
+	Status ParallelStatus `json:"status,omitempty"`
+}
+
+var (
+	// Check that Parallel can be validated and defaulted.
+	_ apis.Validatable = (*Parallel)(nil)
+	_ apis.Defaultable = (*Parallel)(nil)
+
+	// Check that Parallel can return its spec untyped.
+	_ apis.HasSpec = (*Parallel)(nil)
+
+	// TODO: make appropriate fields immutable.
+	//_ apis.Immutable = (*Parallel)(nil)
+
+	_ runtime.Object = (*Parallel)(nil)
+
+	// Check that we can create OwnerReferences to a Parallel.
+	_ kmeta.OwnerRefable = (*Parallel)(nil)
+)
+
+type ParallelSpec struct {
```

Comment:
is the old v1alpha1 ParallelSpec the same as this one? Should we then inline this one there to remove duplicated code? Same with Sequence? and with their statuses...

Reply:
Yeah, so I was hoping to do this in couple of steps:
 1. Get the types moved to the new API group
 2. Create the controller mods
 3. Decide how the migration story goes

Reply:
Sounds good!

Reply:
have we decided to move Parallel and Sequence to v1beta1? I would prefer to stick with `v1alpha1` for now.

Reply:
I'd be fine in upgrading those as well ...